### PR TITLE
Air polution app

### PR DESCRIPTION
**Pull request by mhnazary**
In this project I did:

- [ ] Stunk, CRA, Jest-redux, router, redux, react-icons

- [ ] Create a components folder with:

File air pollution items and a List of these items for the detail section of the page that fetch data from https://openweathermap.org/api/air-pollution

- [ ] File countries items and a list of these items for the home page
- [ ] Europe countries for the API of air pollution parameters
- [ ] Navbar for the navigation bar of the web app using router
- [ ] Search bar on the home page to filter country
- [ ] Back button function in the navbar to return to the home page
- [ ] Detail button in each country to see the air pollution data of that country (Only available in the arrow button, not the entire section)
- [ ] Redux store and DetailSlice for API fetch and search reducers.
- [ ] Jest test for AirItem.js, CountriesList.js, CountriesItem.js and DetailSlice.js
- [ ] Graph bar for the values of air in the country details section.
- [ ] Linter fix
- [ ] Update readme info and deploy/loom video

### CODE REVIEW

File: src/Test/CountryItem.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      <CountriesItem
+        data={{ latitude: 20, longitude: 10 }}
+        imageUrl="image-url"
+        country="Country"
+        capital="Capital"
+      />
+    </MemoryRouter>,
+  );
+
+  const tree = component.toJSON();
```
Comment (Shedrack-Sunday) [inline src/Test/CountryItem.test.js]: - [  ] You did a great job, however, I'm not able to run all your tests successfully. Kindly take a look at the prompt and refactor appropriately 👍 

### View:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @mhnazary          👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the React features ✔️
- Good choice of API and color set ✔️ 
    
    
### Confirmed query:
image

# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key points notes I pointed out to meet the requirements before moving to the next project 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/Test/CountryItem.test.js (hunk 1 above)
Comment (mhnazary) [reply src/Test/CountryItem.test.js]: @Shedrack-Sunday  Thank you for dedicating your valuable time to thoroughly and patiently reviewing my project.
 but, upon executing the test, all the tests passed successfully.

test

File: src/components/CountryItem.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+CountryItem.propTypes = {
+  data: PropTypes.shape({
+    country: PropTypes.string.isRequired,
+    capital: PropTypes.string.isRequired,
+    longitude: PropTypes.number.isRequired,
+    latitude: PropTypes.number.isRequired,
+    image: PropTypes.string.isRequired,
+  }).isRequired,
+};
```
Comment (TadesseDev) [inline src/components/CountryItem.jsx:37]: - [ ] Dear @mhnazary, Just as the previous reviewer mentioned, one of the tests is falling. and the issue is you have not passed the required properties to the test data parameter. As you can see here `country, capital, longitude, latitude, image` are required fields, but in your test, you only pass `latitude: 20, longitude: 10`. Please add the remaining properties to the test component and your test will be fine.

Screen Shot 2023-09-06 at 1 34 22 AM

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @mhnazary 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] You are almost done 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bienvenuushindi — APPROVED** [review]
## Status: Approved :heavy_check_mark: :heavy_check_mark:

Hi @mhnazary     

Your project is complete! Great job for making all the required changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- No linter errors
- All the tests are passing :heavy_check_mark: 
<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**